File: RepoTransBench/target_projects/Java/C#/mar-v-in_ArtHook/tests/original/UtilsTest.cs
using System;
using Xunit;
using ArtHook;

namespace ArtHook
{
    public static class DummyMain
    {
        public static bool Called = false;
        public static Exception? Throwable = null;

        public static void main(string[] args)
        {
            Called = true;
            if (Throwable != null)
                throw Throwable;
        }
    }
}

namespace ArtHook.Tests.Original
{
    public class UtilsTest
    {
        [Fact]
        public void TestCallMainSuccessful()
        {
            ArtHook.DummyMain.Called = false;
            ArtHook.DummyMain.Throwable = null;
            Utils.callMain(typeof(ArtHook.DummyMain).FullName!);
            Assert.True(ArtHook.DummyMain.Called);
        }

        [Fact]
        public void TestCallMainThrowsTargetException()
        {
            ArtHook.DummyMain.Called = false;
            ArtHook.DummyMain.Throwable = new Exception("test");
            Assert.Throws<Exception>(() => Utils.callMain(typeof(ArtHook.DummyMain).FullName!));
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/mar-v-in_ArtHook/tests/original/utils_test.go
package original

import (
    "errors"
    "testing"

    "github.com/stretchr/testify/assert"
)

type DummyMain struct {
    Called    *bool
    Throwable *error
}

// Simulate static method main behavior.
func (d *DummyMain) Main(args []string) error {
    *d.Called = true
    if d.Throwable != nil && *d.Throwable != nil {
        return *d.Throwable
    }
    return nil
}

// Simulate Utils.callMain implementation.
func UtilsCallMain(dummy *DummyMain, className string, args []string) error {
    // In real: use reflection by className. Here just call dummy.Main.
    return dummy.Main(args)
}

func TestUtilsCallMainSuccessful(t *testing.T) {
    called := false
    // No error
    dummy := DummyMain{Called: &called, Throwable: nil}
    err := UtilsCallMain(&dummy, "DummyMain", []string{})
    assert.True(t, called, "DummyMain.main should have been called")
    assert.NoError(t, err)
}

func TestUtilsCallMainThrowsTargetException(t *testing.T) {
    called := false
    throwErr := errors.New("test")
    dummy := DummyMain{Called: &called, Throwable: &throwErr}
    err := UtilsCallMain(&dummy, "DummyMain", []string{})
    assert.True(t, called, "DummyMain.main should have been called")
    assert.Error(t, err)
    assert.EqualError(t, err, "test")
}